### PR TITLE
[CSM] Fix syscall dirty flag after execve

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR fixes the syscall "dirty" flag after an execve and adds functional tests to make sure we don't forget why it's there.

### Motivation

Before this fix, some syscall drift events could have an empty list of syscalls.

### CODE REVIEW

File: pkg/security/tests/security_profile_test.go [hunk 1]
```diff
@@ -2074,3 +2077,466 @@ func TestSecurityProfilePersistence(t *testing.T) {
...
+					for _, s := range expectedSyscalls {
+						if !slices.Contains(event.Syscalls.Syscalls, s) {
+							t.Logf("A ExecveReason syscall drift event was received with the wrong list of syscalls. Expected %v, got %v", expectedSyscalls, event.Syscalls.Syscalls)
+							return false
+						}
+					}
+					if len(event.Syscalls.Syscalls) != len(expectedSyscalls) {
+						t.Logf("A ExecveReason syscall drift event was received with additional syscalls. Expected %v, got %v", expectedSyscalls, event.Syscalls.Syscalls)
+						return false
+					}
```
Comment (spikat) [inline pkg/security/tests/security_profile_test.go:2238]: Could we create an helper for this code (as it's repeated 6 times)?

**Review by spikat — COMMENTED** [review]
some nit comments, otherwise LGTM :)

File: pkg/security/tests/security_profile_test.go [hunk 2]
```diff
@@ -2074,3 +2077,466 @@ func TestSecurityProfilePersistence(t *testing.T) {
...
+			ImageTag:  "latest",
+		},
+		ProfileContexts: map[string]*dumpsv1.ProfileContext{
+			"latest": {
+				Syscalls: []uint32{
+					5,   // SysFstat
+					10,  // SysMprotect
+					11,  // SysMunmap
+					12,  // SysBrk
+					13,  // SysRtSigaction
```
[the remark this answers is not in the document]
Comment (Gui774ume) [reply pkg/security/tests/security_profile_test.go:2094]: Done 👍🏻

**Review by spikat — APPROVED** [review]
LGTM

**Pull request merged**